psi4/src/psi4/dfocc/manager.cc

```diff
@@ -192,8 +192,15 @@ void DFOCC::omp2_manager() {
     Process::environment.globals["SOS-MP2 CORRELATION ENERGY"] = Esosmp2 - Escf;
     Process::environment.globals["SCSN-MP2 CORRELATION ENERGY"] = Escsnmp2 - Escf;
 
-    variables_["MP2 OPPOSITE-SPIN CORRELATION ENERGY"] = Emp2AB;
-    variables_["MP2 SAME-SPIN CORRELATION ENERGY"] = Emp2AA + Emp2BB;
+    if (reference_ == "UNRESTRICTED") {
+        variables_["MP2 OPPOSITE-SPIN CORRELATION ENERGY"] = Emp2AB;
+        variables_["MP2 SAME-SPIN CORRELATION ENERGY"] = Emp2AA + Emp2BB;
+        variables_["MP2 DOUBLES ENERGY"] = Emp2AB + Emp2AA + Emp2BB;
```

Comment:
Are SS/OS still set, but somewhere else? Restricted singles is always zero, using this you may be able to simplify the logic.

Reply:
The way RHF is computed doesn't separate the SS/OS components, so it's a contract violation (https://github.com/psi4/psi4/pull/1850/files#diff-ac285870f28f146a56a93bd3592ff776R400). @JonathonMisiewicz, you fixed this in occ, or am I imagining things?

Reply:
I can't answer, as I can't tell what the two of you are talking about. Daniel seems to be asking if the same-spin and opposite-spin variables are set in response to the lines where they are set, and then asking about the singles energy for some reason. You seem to be talking about whether SCS and SOS is even definable for RHF... Are you proposing we get rid of spin component scaling for closed-shell systems? If that's what you mean, I didn't do that in `occ`, and I don't understand why you would even want it done.

I think I'm missing something here.

Reply:
Looks like I was mistaken about occ and changes on your part, @JonathonMisiewicz.

The `SPIN CORRELATION` variables are only set here in dfocc for U/RO b/c the pieces aren't around for RHF https://github.com/psi4/psi4/blob/master/psi4/src/psi4/dfocc/mp2_direct.cc#L59 . I don't like that situation here, but I'm backing away from this yak. Fortunately this inconsistency doesn't show up by default b/c dfmp2 is the default module.

Reply:
The way I understood Daniel's question was whether SS and OS energies are set for `RESTRICTED` reference elsewhere somewhere. Because they're meaningful quantities for closed shell molecules too, right? In any case, this block is repeated a few times, perhaps it wouldn't hurt to have a helper function?

Reply:
the SS & OS are definitely meaningful quantities that I'd like to collect. however, the formulation in the module doesn't separate them out. Yeah, dfocc could use a lot in the way of helper functions -- I'm just not facing that right now.